plugins/net_plugin/net_plugin.cpp

```diff
@@ -339,9 +342,36 @@ namespace eosio {
    constexpr uint32_t packed_transaction_which = fc::get_index<net_message, packed_transaction>(); // see protocol net_message
 
    class connections_manager {
+   public:
+      struct connection_details {
+         std::string host;
+         connection_ptr c;
+         tcp::endpoint active_ip;
+         tcp::resolver::results_type ips;
+         operator const connection_ptr&() const { return c; }
+      };
+
+      using connection_details_index = multi_index_container<
+         connection_details,
+         indexed_by<
+            ordered_non_unique<
+               tag<struct by_host>,
+               key<&connection_details::host>
+            >,
+            ordered_unique<
+               tag<struct by_connection>,
+               key<&connection_details::c>
+            >,
+            ordered_non_unique<
+               tag<struct by_active_ip>,
```

Comment:
It doesn't look like this index is actually used. The single use could just use `by_host`.

Reply:
Removed.  I had intended to use this index as part of a more sophisticated response to the `/net/disconnect` API, allowing not just perfect string matches for disconnects but also by IP and port, since that's commonly what's available from, for instance, system firewall state tables.